vllm/platforms/cuda.py

```diff
@@ -41,6 +46,42 @@
 torch.backends.cuda.enable_cudnn_sdp(False)
 
 
+@cache
+def _get_backend_priorities(
+    device_capability: DeviceCapability | None = None,
+) -> dict[_Backend, int]:
+    """Get backend priorities with lazy import to avoid circular dependency."""
+    from vllm.attention.backends.registry import _Backend
+
+    if device_capability >= DeviceCapability(10, 0):
+        return {
+            # non-MLA backends
+            _Backend.FLASHINFER: 0,
+            _Backend.FLASH_ATTN: 1,
+            _Backend.TRITON_ATTN: 2,
+            _Backend.FLEX_ATTENTION: 3,
+            # MLA backends
+            _Backend.CUTLASS_MLA: 0,
+            _Backend.FLASHINFER_MLA: 1,
+            _Backend.FLASHMLA: 2,
+            _Backend.FLASH_ATTN_MLA: 3,
+            _Backend.TRITON_MLA: 4,
+        }
```

Comment:
Oh I think we might want this special case to just be for `device_capability == DeviceCapability(10, 0)`. We only support for trtllm attention for sm100 (which is why we want FLASHINFER by default) and CUTLASS_MLA is only an option for sm100.
I'm not sure if we need to change the defaults for sm120 at all from the "normal" case